docker-images/data-refresher/load.py

```diff
@@ -32,22 +51,28 @@ def track_envelope_data_moving_through_dcp(submission_id, failures={}):
         matrix_dynamo_agent.write_item_to_dynamo(matrix_project_payload)
         azul_dynamo_agent.write_item_to_dynamo(azul_project_payload)
     except Exception as e:
-        failures[submission_id] = e
-        print(f"Submission_id {submission_id} failed with error: {e} ")
+        failures[envelope.submission_id] = e
+        print(f"Submission_id {envelope.submission_id} failed with error: {e} ")
+
+
+def _is_excluded_project(project):
+    for exclude_name in PROJECT_NAME_STRINGS_TO_EXCLUDE_FROM_TRACKER:
+        if exclude_name in project.short_name:
+            return True
+    return False
 
 
 def main():
     failures = {}
-    ingest_agent = IngestAgent()
-    primary_envelopes = ingest_agent.get_all_primary_submission_envelopes()
+    ingest_agent = IngestApiAgent(deployment=DEPLOYMENT_STAGE)
     pool = ThreadPool()
-    for envelope in primary_envelopes:
-        submission_id = ingest_agent.get_submission_id_from_envelope(envelope)
-        pool.add_task(track_envelope_data_moving_through_dcp, submission_id, failures)
+    for envelope in SubmissionEnvelope.iter_submissions(ingest_api_agent=ingest_agent, page_size=1000):
```

Comment:
Note there is a little speedup here.  I don't wait until every submission has been collected before I start kicking off data gathering - gathering starts as soon as the first page of results is returned.  I also dropped the check for primary and excluded down into track_envelope_data_moving_through_dcp so it all happens in the same thread.  There is only one thread pool now.

Reply:
Nice!